pkg/download/downloader_test.go

```diff
@@ -55,6 +95,28 @@ func Test_extractZIP(t *testing.T) {
 				"/foo",
 			},
 		},
+		SymlinkTest1(".zip"),
+		SymlinkTest2(".zip"),
+		SymlinkTest3(".zip"),
+	}
+
+	if err := isSymlinkAllowed("foo/bar", "filename", "foo/bar/baz"); err != nil {
+		t.Fatalf("allowed enclosed symlink was disallowed")
+	}
+	if err := isSymlinkAllowed("foo/bar", "..", "foo/bar/baz/bam"); err != nil {
+		t.Fatalf("allowed symlink escape to staging area was disallowed")
+	}
+	if err := isSymlinkAllowed("foo/bar", "../../", "foo/bar/baz/bam"); err == nil {
+		t.Fatalf("disallowed symlink escape was allowed")
+	}
+	if err := isSymlinkAllowed("foo/bar", "gorp/../..", "foo/bar/baz/bam"); err != nil {
+		t.Fatalf("allowed symlink escape was disallowed")
+	}
+	if err := isSymlinkAllowed("foo/bar", "gorp/../../..", "foo/bar/baz/bam"); err == nil {
+		t.Fatalf("disallowed symlink escape was allowed")
+	}
+	if err := isSymlinkAllowed("foo/bar", "/absolute", "foo/bar/baz/bam"); err == nil {
+		t.Fatalf("disallowed symlink absolute path escape was allowed")
```

Comment:
you can keep these but these are better suited for a `TestIsSymlinkAllowed` method.

Reply:
reading again, these don't seem to have anything to do with the rest of this test method. let's move out to unit-test.